File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression_me.py
# Simple Linear Regression

# importing libraries and packages
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# data pre-processing phase
dataset = pd.read_csv('Salary_Data.csv')

# creating matrix X and vector y
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Converting whole dataset into training and testing set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# there is no need for feature scaling as the model we will be 
# using further for linear regression will take care for scaling.

# fitting the linear regression model to training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)


# predicting the test set results
y_pred = regressor.predict(X_test)

# visualizing data points and simple linear regression model
# Visulaizing Training set results
plt.scatter(X_train, y_train, color = "red")
plt.plot(X_train, regressor.predict(X_train), color = "blue")
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Experience')
plt.ylabel('Salary')
plt.show()

# Visualizing Test Set results
plt.scatter(X_test, y_test, color = "red")
plt.plot(X_test, regressor.predict(X_test), color = "blue")
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Experience')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression_me.R
# Simple Linear Regression

# data pre-processing phase

# importing dataset
dataset = read.csv('Salary_Data.csv')

# splitting data into training and test set
library(caTools)
set.seed(1234)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# fitting simple linear regressor model to training set
regressor  = lm(formula = Salary ~ YearsExperience,
                data = training_set)

# predicting test set results
y_pred = predict(regressor, newdata = test_set)

# installing ggplot2 for visualization
# install.packages('ggplot2')
library(ggplot2)

# Visualizing training set results
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), 
             colour = "red") +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, training_set)), 
            colour = "blue") + 
  ggtitle('Salary vs Experience (Training Set)') + 
  xlab('Years of Experience') +
  ylab('Salary')

# Visualizing test set results
ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
             colour = "red") + 
  geom_line(aes(x = test_set$YearsExperience, y = predict(regressor, test_set)),
            colour = "blue") + 
  ggtitle('Salary vs Experience (Test Set)') + 
  xlab('Years of Experience') + 
  ylab('Salary')

